api/src/main/java/org/pmiops/workbench/auth/UserProvider.java

```diff
@@ -0,0 +1,18 @@
+package org.pmiops.workbench.auth;
+
+import javax.inject.Provider;
+import org.pmiops.workbench.db.model.User;
+import org.springframework.stereotype.Service;
+
+@Service
```

Comment:
Per offline discussion, I would definitely favor our previous approach over this. I think this convention changes the production code path too significantly to justify saving some test boilerplate. I don't see a particularly high cost in this PR to justify it.

I didn't get a chance to try the static stubbing approach per slack - alternatively if you can find a way to make that work, that would fix all the problems here.